lib/stats_reporter.rb

```diff
@@ -0,0 +1,98 @@
+# generates text for a report on storage root size and object status
+class StatsReporter
+
+  require 'text-table'
+  require 'open3'
+
+  def self.storage_report_text
+    head = %w[filesystem total used pct_used free pct_free]
+    Text::Table.new(head: head, rows: storage_report_lines).to_s
+  end
+
+  def workflow_report_text
+    head = %w[workflow waiting error recent archived]
+    Text::Table.new(head: head, rows: workflow_report_lines).to_s
+  end
+
+  def self.workflow_xml
+    File.open('config/workflows/sdr/preservationIngestWF') { |f| Nokogiri::XML(f) }
+  end
+
+  def self.repository
+    workflow_xml.at_xpath('//workflow-def/@repository').value
+  end
+
+  def self.ingest_wf
+    workflow_xml.at_xpath('//workflow-def/@id').value
+  end
+
+  def self.ingest_wf_steps
+    workflow_xml.xpath('//process/@name').map(&:value)
+  end
+
+  def self.df_output
+    stdout_str, _status = Open3.capture2('df -h')
+    stdout_str
+  end
+
+  def self.storage_mount_lines
+    df_output.split("\n").select { |line| line.include?('services-disk') }
+  end
+
+  def self.parsed_storage_mount_lines
+    storage_mount_lines.map do |line|
+      storage_root = {}
```

Comment:
No reason not to do some (or all) of your assignment at the same time you are declaring the new hash.  Like:
```ruby
storage_root = {
  total: split_line[0],
  used: split_line[1],
  remaining: split_line[2],
  percent_used: split_line[3],
  filesystem: split_line[4] # ...
}
```

Reply:
(And if you do it all in one go, then you don't need the variable assignment at all.)

Reply:
My example above can be further refined to a single line:
```ruby
storage_root = [:total, :used, :remaining, :percent_used, :filesystem].zip(split_line[0..4]).to_h
```

Reply:
That further refinement works until `:percent_free`, which has a more complicated construction I'd vote for the former, which I like for keeping everything in one declaration.